### DISCUSSION TITLE
Use highlight instead of jinx-highlight on mouse overlay

### DISCUSSION THREAD

**Discussion by ghost** [minad/jinx#184] [Q&A] [answered]
Locally, i have `(put 'jinx-overlay 'mouse-face '(jinx-misspelled highlight))`.  I think this is the better default.  Almost everywhere else in emacs, the highlight face is used for interactive text under the mouse.  Using the `jinx-highlight` face here isn't ideal because it acts differently to everywhere else.

I can think of three ways to change this behaviour.

1. Use `highlight` instead of `jinx-highlight` for the mouse overlay.  This is the most simple, but as you've chosen to use `jinx-highlight` it seems you might have a reason to keep this separate from `highlight`.
2. Create a new face, eg `jinx-mouse-highlight`.  This would work well for me, i can just change the face to look how i like it.  By default, you could keep it indistinguishable from `jinx-highlight`.
3. Create a custom variable, as goto-address mode does, something like `jinx-highlight-mouse-face`.  By default you could keep this is `jinx-highlight`, but i could easily set it to `highlight` or anything else.

**Comment by minad** [comment, marked as answer]
The `jinx-highlight` face is used when correcting misspellings with `jinx-correct`. Therefore it feels a little more consistent within Jinx to use the face also for corrections with the mouse. But if you prefer to change the face, the right thing to do is `(put 'jinx-overlay 'mouse-face '(jinx-misspelled highlight))` as you have done. I've set the Jinx overlay category especially to allow overrides by the user.

**Reply by ghost** [reply]
Thanks.  I'm pretty happy with this solution, it's simple and works.
